userportaldatamodel/search.py

```diff
@@ -52,29 +54,44 @@ class ProjectSearch(Base):
     search = relationship("Search", backref=backref("project_has_search"))
 
     create_date = Column(DateTime(timezone=False), server_default=func.now())
-    update_date = Column(DateTime(timezone=False), server_default=func.now(), onupdate=func.now())
+    update_date = Column(
+        DateTime(timezone=False), server_default=func.now(), onupdate=func.now()
+    )
 
 
 # class ProjectSearch(Base):
 #     __tablename__ = 'project_has_search'
 #     project_id = Column(
-#         Integer, 
-#         ForeignKey('project.id'), 
+#         Integer,
+#         ForeignKey('project.id'),
 #         primary_key = True)
 
 #     search_id = Column(
-#         Integer, 
-#         ForeignKey('search.id'), 
+#         Integer,
+#         ForeignKey('search.id'),
 #         primary_key = True)
 
+class SearchIsShared(Base):
+    """A search that is shared with another user."""
+
+    __tablename__ = "search_is_shared"
+    search_id = Column(Integer, ForeignKey("search.id"), primary_key=True)
+    user_id = Column(
```

Comment:
If this is nullable, which I agree with, it shouldn't be a primary. However it should be a foreign key

Reply:
Right. Good point!